### PR TITLE
Wifi Selection UI

### PR DESCRIPTION
**Pull request by rmescandon**
Improved Look and feel of the available WIFIs list:
- Using vanilla framework theme
- Added feature to select an available network, provide a password, toggle show/hide that password and connect or cancel
- Updated README.md with needed steps for further development in this web UI

Known issues:
After selecting a network, provide password and click to connect, no matter the result we won't receive any reply of the network connect operation result. This happens because that reply is async to the operation triggering. 
- One solution could be having the user to refresh the UI
- Other solution could be requesting for the network state by polling and and show intermediate and final state. 
- And other, maybe the better solution, could be using websockets to get connection changes from the backend side directly in the UI. (for this that would be maybe better having a React.js or other fw solution other than golang direct templates as we have so far, as things get complicated...)

No matter which solution is taken, for this to work properly it is needed including in netman package operation to know current status of the different interfaces, so that the list of available ones can be shown in different colors depending on their current status, and also having signals when certain interface changes its status.

from /org/freedesktop/NetworkManager/Devices/<id> in system dbus it is possible request property org.freedesktop.NetworkManager.Device.State which would return its current state (https://developer.gnome.org/NetworkManager/unstable/nm-dbus-types.html#NMDeviceState). The ones we are at least interested are:
NM_DEVICE_STATE_DISCONNECTED = 30
NM_DEVICE_STATE_CONFIG = 50
NM_DEVICE_STATE_ACTIVATED = 100

that would be set respectively to disconnected, connecting and connected in the UI.

All this can be implemented in another PR. (I think)

### CODE REVIEW

**Comment by knitzsche** [comment]
Hi Robert. Regarding the issue you describe, it is not expected that the user will be able see the response to the connection request because connecting to the new external AP also involves first bringing down the current wifi-ap through which the user is connected to the portal. This is because (for version 1 anyway), it is not expected that we can simultaneously have the wifi-ap UP and connected to the user and connect to an external AP through the sam single wifi card.

**Comment by knitzsche** [comment]
Indeed, the portal should have some text that informs the user that their current connection will be lost (when they click OK) and they should try to reconnect to the device over the new AP connection after a minute or so. (Whether they will connect via the new IP address (which they would need to get from their AP I presume) or via avahi (which we need to look into) is an open question.)

**Comment by rmescandon** [comment]
Ah true. You are right.
I need then to remove some not needed network status stuff and also add that inform text when pressed connect button.
I'll let you know when ready again to review

File: server/handlers.go [hunk 1]
```diff
@@ -19,15 +19,15 @@ var ResourcesPath = filepath.Join(os.Getenv("SNAP"), "static")
 
 // PageData dynamic data to fulfill the template
 type PageData struct {
-	Ssids []netman.SSID
+	Networks []netman.SSID
```
Comment (knitzsche) [inline server/handlers.go]: Quibble, but can we keep the name Ssids (not Networks)? Its more *wifi* specific.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,51 @@
...
+# Verify snapcraft is installed ok by printing out the version
+snapcraft -v
+2.27.1
+```
+should output current version. More information on [snapcraft.io](https://snapcraft.io)
+
+### Install Go
+Follow the instructions to [install Go](https://golang.org/doc/install).
+
+- Install NVM
```
Comment (knitzsche) [inline README.md]: markdown formatting seems amiss: https://private-fileshare.canonical.com/~knitzsche/sol-eng/wifi-connect/PR-17/r.html

File: static/templates/ssids.html [hunk 3]
```diff
@@ -4,29 +4,103 @@
...
+                                <input type="password" id="password{{$i}}"/>
+                            </li>
+                            <li>
+                                <input type="checkbox" id="showpassword{{$i}}" onchange="show_password({{$i}})"/>
+                                <label for="showpassword">show password</label>
+                            </li>
+                            <li>
+                                 <div id="alert{{$i}}" class="cheshire box" style="background-color: #eee">
+                                    <h3>Caution!</h3>
+                                     <p>While connecting to an external network you won't be able to manage device. Proceed?.</p>
```
Comment (knitzsche) [inline static/templates/ssids.html]: I prefer a something a little more direct for now: "Continuing will disconnect you from the device by taking down its wifi network. This page will be unavailable. After continuing, you may connect to the device through its new wifi connection."

File: static/templates/ssids.html [hunk 4]
```diff
@@ -4,29 +4,103 @@
...
+                <tr>
+                    <th scope="row">
+                        <label class="collapse" for="radio{{$i}}">{{$network.Ssid}}</label>
+                        <input id="radio{{$i}}" type="radio" name="c1">
+                        <div class="six-col">
+                        <fieldset>
+                        <ul class="no-bullets">
+                            <li>
+                                <input type="hidden" id="ssid{{$i}}" value="{{$network.Ssid}}"/>
+                                <input type="password" id="password{{$i}}"/>
```
Comment (knitzsche) [inline static/templates/ssids.html]: Please add label above password box saying "Enter passphrase:"

File: static/templates/ssids.html [hunk 5]
```diff
@@ -4,29 +4,103 @@
...
+                            </li>
+                            <li>
+                                <input type="checkbox" id="showpassword{{$i}}" onchange="show_password({{$i}})"/>
+                                <label for="showpassword">show password</label>
```
Comment (knitzsche) [inline static/templates/ssids.html]: change to "show passphrase"

File: static/templates/connect_result.html [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    <link rel="stylesheet" href="/static/css/application.css" />
+</head>
+
+<body>
+    <div class="wrapper">
+        <div id="main-content" class="inner-wrapper">
+            <div class="row no-border" id="grid">
+            <h3>Connecting...</h3>
+                 <div class="twelve-col box" style="background-color: #eee">
+                    <p>Network will be available if provided credentials are valid. Please check device connection after passed some minutes</p>
```
Comment (knitzsche) [inline static/templates/connect_result.html]: Please change to: 
Device is now attempting to connect to <SSID>. 
You may try to connect to the device through the new <SSID> connection.

File: server/handlers.go (hunk 1 above)
Comment (rmescandon) [inline server/handlers.go]: done

File: static/templates/ssids.html (hunk 4 above)
Comment (rmescandon) [inline static/templates/ssids.html]: done

File: static/templates/ssids.html (hunk 5 above)
Comment (rmescandon) [inline static/templates/ssids.html]: done

File: README.md (hunk 2 above)
Comment (rmescandon) [inline README.md]: mmm.. browsers by default don't render markdown. Github does. You can see it at https://github.com/rmescandon/UCWifiConnect/blob/look-and-feel/README.md.
Also, If you use Visual Studio Code as IDE, you can switch between edit and view views by pressing Ctrl+Alt+V.
In any case, it is true that i missed one header for "web development" section, just added

File: static/templates/connect_result.html (hunk 6 above)
Comment (rmescandon) [inline static/templates/connect_result.html]: Done. 
I've also renamed template connect_result.html to connecting.html, and refactor a bit the internal code in server/handler.go as we now need to execute both templates with data

File: static/templates/ssids.html (hunk 3 above)
Comment (rmescandon) [inline static/templates/ssids.html]: done

**Comment by rmescandon** [comment]
Ready to re-review

File: README.md (hunk 2 above)
Comment (knitzsche) [inline README.md]: I created the html file with markdown command...

**Pull request merged**